### PR TITLE
feat: support Casdoor storage API

### PR DESCRIPTION
**Pull request by wintbiit**
solved #584 

added abstract [storage.go](https://github.com/WinterOfBit/casibase/blob/feature/storage/storage/storage.go) and impl casdoor.go

tested using [storage_test#TestCasdoor()](https://github.com/WinterOfBit/casibase/blob/8c0ae44eaf963f82fa9fc85c192a1584263fc1c5/storage/storage_test.go#L35)

### CODE REVIEW

**Comment by hsluoyz** [comment]
@WinterOfBit plz fix linter and CLA:

image

**Comment by WinterOfBit** [comment]
> @WinterOfBit plz fix linter and CLA:
> 
> image

fixed

File: storage/casdoor.go [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+		return nil, err
+	}
+
+	return response.Body, nil
+}
+
+func (s *casdoorClient) Put(user, key string, bytes []byte) error {
+	_, _, err := casdoorsdk.UploadResource(user, "Casibase", "Casibase",
+		util.GetIdFromOwnerAndName(fmt.Sprintf("/resource/%s/%s/casibase",
+			beego.AppConfig.String("casdoorOrganization"),
```
Comment (hsluoyz) [inline storage/casdoor.go]: Define some shared global variables, don't use `beego.AppConfig.String("xxx")` so many times

File: storage/storage.go [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+	List(prefix string) ([]*Object, error)
+}
+
+type Object struct {
+	Key          string
+	LastModified *time.Time
+	Storage      Storage
+}
+
+func NewStorageProvider(provider string) Storage {
```
Comment (hsluoyz) [inline storage/storage.go:21]: Is it possible to select different Casdoor providers to use for each Casibase store? A provider name will be specified in the store edit page

**Review by hsluoyz — CHANGES_REQUESTED** [review]
Also demo video for non-aliyun provider?

**Comment by wintbiit** [comment]
> Also demo video for non-aliyun provider?

working on frontend impl

**Comment by hsluoyz** [comment]
@WinterOfBit hi, any update?

**Review by hsluoyz — APPROVED** [review]

**Comment by wintbiit** [comment]
> @WinterOfBit hi, any update?

@hsluoyz Being busy for some days. I'll make another pr later to make this func more perfect.

**Comment by hsluoyz** [comment]
@WinterOfBit the whole task has been finished already :) no need to worry now

**Comment by wintbiit** [comment]
> @WinterOfBit the whole task has been finished already :) no need to worry now

@hsluoyz ok. Glad to see that!

**Pull request merged**